src/main/java/uk/ac/ebi/ena/readtools/fastq/ena/Fastq2BamConsumer.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2010-2020 EMBL - European Bioinformatics Institute
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package uk.ac.ebi.ena.readtools.fastq.ena;
+
+import htsjdk.samtools.ReservedTagConstants;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import uk.ac.ebi.ena.readtools.common.reads.QualityNormalizer;
+import uk.ac.ebi.ena.readtools.loader.common.InvalidBaseCharacterException;
+import uk.ac.ebi.ena.readtools.loader.common.consumer.DataConsumable;
+import uk.ac.ebi.ena.readtools.loader.common.consumer.DataConsumer;
+import uk.ac.ebi.ena.readtools.loader.common.consumer.DataConsumerException;
+import uk.ac.ebi.ena.readtools.loader.fastq.IlluminaSpot;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Accepts Illumina spot data and writes them out to a BAM file.
+ */
+public class Fastq2BamConsumer implements DataConsumer<IlluminaSpot, DataConsumable> {
+
+    private static final String DEFAULT_READ_GROUP_NAME = "A";
+
+    private static final String VALID_DNA_CHARSET = "^[.acmgrsvtwyhkdbnNACMGRSVTWYHKDBN]+$";
+    private static final Pattern VALID_DNA_CHARSET_PATTERN = Pattern.compile(VALID_DNA_CHARSET);
+
+    private final QualityNormalizer qualityNormalizer;
+
+    private final String sampleName;
+
+    private final SAMFileWriter writer;
+
+    private volatile boolean isOk = true;
+
+    public Fastq2BamConsumer(QualityNormalizer qualityNormalizer, String sampleName, String outputFilePath) {
+        this.qualityNormalizer = qualityNormalizer;
+        this.sampleName = sampleName;
+
+        if (sampleName == null || sampleName.trim().isEmpty()) {
+            throw new IllegalArgumentException("Sample name is either null or empty.");
+        }
+
+        writer = new SAMFileWriterFactory().makeSAMOrBAMWriter(
+                createHeader(), false, Paths.get(outputFilePath));
+    }
+
+    @Override
+    public void consume(IlluminaSpot iSpot) throws DataConsumerException {
+        try {
+            Matcher matcher = VALID_DNA_CHARSET_PATTERN.matcher(iSpot.bases);
+            if( !matcher.matches() )
+                handleInvalidDnaCharset(iSpot.bases, matcher);

```

Comment:
Generic comment. Please use Google Java format that e.g. always requires parenthesis after if.